cclib/method/bader.py

```diff
@@ -0,0 +1,223 @@
+                                    yindex + nextDirection[1],
+                                    zindex + nextDirection[2],
+                                ]
+                                != 0
+                            ):
+                                # Path arrived at a point that has already been assigned with an index
+                                baderIndex = self.fragresults[
+                                    xindex + nextDirection[0],
+                                    yindex + nextDirection[1],
+                                    zindex + nextDirection[2],
+                                ]
+                                listcoord = tuple(numpy.array(listcoord).T)
+                                self.fragresults[listcoord] = baderIndex
+
+                                loopcondition = False
+
+                            elif nextDirection == [0, 0, 0]:
+                                # This point is local maximum
+                                # Assign this point and everything along the path with a new index
+                                baderIndex = nextIndex
+                                nextIndex += 1
+
+                                listcoord = tuple(numpy.array(listcoord).T)
+                                self.fragresults[listcoord] = baderIndex
+
+                                loopcondition = False
+
+                            elif (
+                                xindex + nextDirection[0] == 0
+                                or xindex + nextDirection[0] == (len(self.chgdensity.data) - 1)
+                                or yindex + nextDirection[1] == 0
+                                or yindex + nextDirection[1] == (len(self.chgdensity.data[0]) - 1)
+                                or zindex + nextDirection[2] == 0
+                                or zindex + nextDirection[2]
+                                == (len(self.chgdensity.data[0][0]) - 1)
+                            ):
+                                # Avoid exiting the grid
+                                baderIndex = nextIndex
+                                nexIndex += 1
+
+                                listcoord = tuple(np.array(listcoord).T)
+                                self.fragresults[listcoord] = baderIndex
+
+                                loopcondition = False
+
+                            else:
+                                # Advance to the next point according to the direction of
+                                # maximum gradient
+                                xindex += nextDirection[0]
+                                yindex += nextDirection[1]
+                                zindex += nextDirection[2]
+
+        # Now try to identify each Bader region to individual atom.
+        # Try to find an area that captures enough representation
+        self.matches = numpy.zeros_like(self.data.atomnos)
+        for pos in range(len(self.data.atomcoords[-1])):
+            gridpt = numpy.round(
```

Comment:
I think you can replace `gridpt` with `xgrid, ygrid, zgrid` and also have `dtype=int` to avoid a separate conversion.

Reply:
Yes. I've just replaced it to make it simpler and more readable. Thanks.

Reply:
I've changed this part of the code back -- because indices to the array should not be numpy.ndarray, I think replacing gridpt with xgrid, ygrid, zgrid will still require separate line of int() conversion. Let me know if there is a way that is simpler.

Reply:
For some reason I can't figure out how to do this cleanly, so it's fine as it currently is.